prompb/rwcommon/codec_test.go

```diff
@@ -185,22 +190,29 @@ func testFloatHistogram() histogram.FloatHistogram {
 			{Offset: 0, Length: 1},
 		},
 		NegativeBuckets: []float64{1, 3, 1, 2, 1, 1},
+		CustomValues:    []float64{21421, 523},
 	}
 }
 
 func TestFromIntToFloatOrIntHistogram(t *testing.T) {
-	testIntHist := testIntHistogram()
-	testFloatHist := testFloatHistogram()
-
 	t.Run("v1", func(t *testing.T) {
-		h := prompb.FromIntHistogram(123, testIntHist.Copy())
+		// v1 does not support nhcb.
+		testIntHistWithoutNHCB := testIntHistogram()
+		testIntHistWithoutNHCB.CustomValues = nil
+		testFloatHistWithoutNHCB := testFloatHistogram()
+		testFloatHistWithoutNHCB.CustomValues = nil
+
+		h := prompb.FromIntHistogram(123, &testIntHistWithoutNHCB)
 		require.False(t, h.IsFloatHistogram())
 		require.Equal(t, int64(123), h.Timestamp)
-		require.Equal(t, testIntHist, *h.ToIntHistogram())
-		require.Equal(t, testFloatHist, *h.ToFloatHistogram())
+		require.Equal(t, testIntHistWithoutNHCB, *h.ToIntHistogram())
```

Comment:
nit: in theory this can pass if `prompb.FromIntHistogram(timestamp int64, h *histogram.Histogram)` sets all fields to nil in `h`. So you'd want to either pass a copy to `prompb.FromIntHistogram` or just not reuse `testIntHistWithoutNHCB` in the assertion , rather create a new instance.  In reality there's enough redundancy to stop it from happening, so this is ok.

Reply:
Yes, but I would not be too rigid here, you can check this 20 line of function that it does not change argument, just it uses pointer for efficiency & consistency (passing as copy would still reuse some memory e.g. arrays).